plugins/kotlin-dataframe/src/org/jetbrains/kotlinx/dataframe/plugin/impl/api/groupBy.kt

```diff
@@ -199,4 +210,217 @@ abstract class GroupByAggregator(val defaultName: String) : AbstractSchemaModifi
 }
 
 class GroupByMaxOf : GroupByAggregator(defaultName = "max")
+
 class GroupByMinOf : GroupByAggregator(defaultName = "min")
+
+abstract class GroupByAggregatorExpressionMean(val defaultName: String) : AbstractSchemaModificationInterpreter() {
+    val Arguments.receiver by groupBy()
+    val Arguments.name: String? by arg(defaultValue = Present(null))
+    val Arguments.expression by type()
+
+    override fun Arguments.interpret(): PluginDataFrameSchema {
+        val aggregated = makeNullable(simpleColumnOf(name ?: defaultName, session.builtinTypes.doubleType.type))
+        return PluginDataFrameSchema(receiver.keys.columns() + aggregated)
+    }
+}
+
+class GroupByMeanOf : GroupByAggregatorExpressionMean(defaultName = "mean")
+
+class GroupByStdOf : GroupByAggregatorExpressionMean(defaultName = "std")
+
+abstract class GroupByAggregatorExpressionComparable(val defaultName: String) :
+    AbstractSchemaModificationInterpreter() {
+    val Arguments.receiver by groupBy()
+    val Arguments.name: String? by arg(defaultValue = Present(null))
+    val Arguments.expression by type()
+
+    override fun Arguments.interpret(): PluginDataFrameSchema {
+
+        val aggregated = makeNullable(simpleColumnOf(name ?: defaultName, createComparableType(session)))
```

Comment:
Looks like it needs to be updated after Jolan's PR
Now runtime type is Comparable<Any?> which seems wrong, because all values are actually Int. We discussed what result type we want to median